### PR TITLE
[MRG] Linear Neural Networks/Concise Implementation of Softmax Regression

### PR DESCRIPTION
**Pull request by xiaotinghe**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-558/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

**Comment by mli** [comment]
Job d2l-zh/PR-558/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

**Comment by mli** [comment]
Job d2l-zh/PR-558/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

File: chapter_linear-networks/softmax-regression-concise.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Softmax 回归的简洁实现
+# Softmax回归的简洁实现
 :label:`sec_softmax_concise`
 
-正如深度学习框架的高级 API 使得在 :numref:`sec_linear_concise` 中实现线性回归变得更加容易一样，我们会发现它类似（或可能更方便）实现分类模型。让我们坚持使用时尚多国主义数据集，并将批量大小保持在 256，如 :numref:`sec_softmax_scratch` 中。
+正如通过深度学习框架的高级 API 能够使得在 :numref:`sec_linear_concise` 中实现线性回归变得更加容易一样，我们会发现它同样能更方便地实现分类模型。让我们继续使用Fashion-MNIST数据集，并保持批量大小为256，就像在:numref:`sec_softmax_scratch` 中一样。
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-concise.md:4]: “正如通过深度学习框架的高级 API 能够使得在 :numref:`sec_linear_concise` 中实现线性回归变得更加容易一样，我们会发现它同样能更方便地实现分类模型”

不通顺

File: chapter_linear-networks/softmax-regression-concise.md [hunk 2]
```diff
@@ -64,16 +64,17 @@ weight_initializer = tf.keras.initializers.RandomNormal(mean=0.0, stddev=0.01)
 net.add(tf.keras.layers.Dense(10, kernel_initializer=weight_initializer))
 ```
 
-## 重新审视软件最大实施
+## 重新审视Softmax的实现
 :label:`subsec_softmax-implementation-revisited`
 
-在前面的 :numref:`sec_softmax_scratch` 样本中，我们计算了模型的输出，然后通过交叉熵损耗运行此输出。从数学上讲，这是一件完全合理的事情。然而，从计算角度来看，指数可能是数值稳定性问题的一个来源。
+在前面 :numref:`sec_softmax_scratch` 的例子中，我们计算了模型的输出，然后此输出通过交叉熵损失。从数学上讲，这是一件完全合理的事情。然而，从计算角度来看，指数可能会造成数值稳定性问题。
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-concise.md:70]: “然后此输出通过交叉熵损失”

不准确

File: chapter_linear-networks/softmax-regression-concise.md [hunk 3]
```diff
@@ -64,16 +64,17 @@ weight_initializer = tf.keras.initializers.RandomNormal(mean=0.0, stddev=0.01)
...
 
-回想一下，软最大函数计算 $\hat y_j = \frac{\exp(o_j)}{\sum_k \exp(o_k)}$，其中 $\hat y_j$ 是预测概率分布的 $j^\mathrm{th}$ 元素和 $o_j$ 是对数的 $j^\mathrm{th}$ 元素。如果 $o_k$ 中的一些非常大（即非常积极），那么 $\exp(o_k)$ 可能大于我们可以拥有的特定数据类型（即 * 溢出 *）的最大数字。这将使分母（和/或分子）`inf`（无穷大），我们最后遇到的是 0，`inf` 或 `nan`（不是数字）的 $\hat y_j$。在这些情况下，我们没有得到一个明确定义的交叉熵返回值。
+回想一下，Softmax函数 $\hat y_j = \frac{\exp(o_j)}{\sum_k \exp(o_k)}$，其中$\hat y_j$是预测的概率分布。$o_j$是未归一化的预测$\mathbf{o}$的第$j^\mathrm{th}$个元素。如果 $o_k$ 中的一些数值非常大，那么 $\exp(o_k)$ 可能大于数据类型容许的最大数字（即 *上溢出*）。这将使分母或分子变为`inf`（无穷大），我们最后遇到的是0、`inf` 或 `nan`（不是数字）的 $\hat y_j$。在这些情况下，我们不能得到一个明确定义的交叉熵的返回值。
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-concise.md:72]: 之前用了“上溢”，为了保持一致，这里不应用“上溢出”

File: chapter_linear-networks/softmax-regression-concise.md [hunk 4]
```diff
@@ -64,16 +64,17 @@ weight_initializer = tf.keras.initializers.RandomNormal(mean=0.0, stddev=0.01)
...
 
-解决这个问题的一个诀窍是先从所有 $o_k$ 中减去 $\max(o_k)$，然后再继续进行软最大计算。您可以验证每个 $o_k$ 按常数因子进行的移动是否不会更改 softmax 的返回值。在减法和归一化步骤之后，可能有些 $o_j$ 具有较大的负值，因此相应的 $\exp(o_j)$ 将接近零的值。由于有限的精度（即 * 下水 *），这些值可能会四舍五入为零，使 $\hat y_j$ 为零，并且给我们 $\log(\hat y_j)$ 的值 `-inf`。在反向传播的道路上，我们可能会发现自己面临着一个可怕的 `nan` 结果的屏幕。
+解决这个问题的一个技巧是，在继续softmax计算之前，先从所有$o_k$中减去$\max(o_k)$。你可以证明每个 $o_k$ 按常数进行的移动不会改变softmax的返回值。在减法和归一化步骤之后，可能有些 $o_j$ 具有较大的负值。由于精度受限， $\exp(o_j)$ 将有接近零的值（即 *下溢出*）。这些值可能会四舍五入为零，使 $\hat y_j$ 为零，并且使得 $\log(\hat y_j)$ 的值为 `-inf`。反向传播几步后，我们可能会发现自己面对一屏幕可怕的`nan`结果。
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-concise.md:74]: 下溢

File: chapter_linear-networks/softmax-regression-concise.md [hunk 5]
```diff
@@ -83,7 +84,8 @@ $$
 \end{aligned}
 $$
 
-如果我们想要通过模型评估输出概率，我们希望保持传统的 softmax 函数方便。但不是将 softmax 概率传递到我们的新损失函数中，我们只是传递日志并计算 softmax 及其日志在交叉熵损失函数中一次性，这做了像 [“LogSumExp 技巧”]（https://en.wikipedia.org/wiki/LogSumExp）这样的聪明事情。
+我们也希望保留传统的softmax函数，以备我们需要评估通过模型输出的概率。
+但是，我们没有将softmax概率传递到损失函数中，而是在交叉熵损失函数中传递未归一化的预测并同时计算softmax及其对数，这是一件聪明的事情 ["LogSumExp trick"](https://en.wikipedia.org/wiki/LogSumExp).
```
Comment (astonzhang) [inline chapter_linear-networks/softmax-regression-concise.md:88]: trick翻译一下

**Comment by mli** [comment]
Job d2l-zh/PR-558/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

**Comment by mli** [comment]
Job d2l-zh/PR-558/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

**Comment by mli** [comment]
Job d2l-zh/PR-558/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/

**Comment by mli** [comment]
Job d2l-zh/PR-558/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-558/
